src/components/Common/CommonList.vue

```diff
@@ -0,0 +1,236 @@
+      :display-create="displayCreate"
+      :all-checked="allChecked"
+      :selected-documents="selectedDocuments"
+      :length-document="selectedDocuments.length"
+      :document-to-delete="documentToDelete"
+      :perform-delete="performDelete"
+      @filters-updated="onFiltersUpdated"
+      @create-clicked="create"
+      @toggle-all="toggleAll"
+      @crudl-refresh-search="fetchDocuments">
+
+        <div class="collection">
+          <div class="collection-item collection-transition" v-for="document in documents" :key="document.id">
+            <component
+              :is="itemName"
+              @checkbox-click="toggleSelectDocuments"
+              :document="document"
+              :is-checked="isChecked(document.id)"
+              :index="index"
+              :collection="collection"
+              @common-list::edit-document="editDocument"
+              @delete-document="deleteDocument">
+            </component>
+          </div>
+        </div>
+
+    </crudl-document>
+  </div>
+</template>
+
+<style lang="scss" rel="stylesheet/scss" scoped>
+.collection {
+  overflow: visible;
+}
+</style>
+
+<script>
+import CrudlDocument from './CrudlDocument'
+import UserItem from '../Security/Users/UserItem'
+import RoleItem from '../Security/Roles/RoleItem'
+import ProfileItem from '../Security/Profiles/ProfileItem'
+import DocumentItem from '../Data/Documents/DocumentItem'
+import {
+  filterManager,
+  NO_ACTIVE,
+  Filter,
+  formatSort,
+  searchFilterOperands
+} from '../../services/filterManager'
+import { SET_TOAST } from '../../vuex/modules/common/toaster/mutation-types'
+
+export default {
+  name: 'CommonList',
+  components: {
+    CrudlDocument,
+    UserItem,
+    RoleItem,
+    ProfileItem,
+    DocumentItem
+  },
+  props: {
+    index: String,
+    collection: String,
+    itemName: String,
+    displayCreate: {
+      type: Boolean,
+      default: false
+    },
+    performSearch: Function,
+    performDelete: Function,
+    routeCreate: String,
+    routeUpdate: String
+  },
+
+  data() {
+    return {
+      searchFilterOperands,
+      selectedDocuments: [],
+      documents: [],
+      totalDocuments: 0,
+      documentToDelete: null,
+      currentFilter: new Filter()
+    }
+  },
+  computed: {
+    isDocumentListFiltered() {
+      return this.currentFilter.active !== NO_ACTIVE
+    },
+    isCollectionEmpty() {
+      return !this.isDocumentListFiltered && this.totalDocuments === 0
+    },
+    displayBulkDelete() {
+      return this.selectedDocuments.length > 0
+    },
+    allChecked() {
+      if (!this.selectedDocuments || !this.documents) {
+        return false
+      }
+
+      return this.selectedDocuments.length === this.documents.length
+    },
+    paginationFrom() {
+      return parseInt(this.currentFilter.from) || 0
+    },
+    paginationSize() {
+      return parseInt(this.currentFilter.size) || 10
+    }
+  },
+  methods: {
+    isChecked(id) {
```

Comment:
This shouldn't be a computed value ?

Reply:
No. This is a function that returns true if the document with the given id is selected. Computed values do not accept arguments.